src/components/course-calendar/plan.cjsx

```diff
@@ -103,190 +106,104 @@ CoursePlan = React.createClass
-    #       stack on top of other plans that overlap in duration.
-    planStyle =
-      width: durationLength * 100 / 7 + '%'
-      left: offset * 100 / 7 + '%'
-      top: (weekTopOffset + 4 - order * 3) + 'rem'
+  removeHover: ->
+    @setState(isHovered: false) if @state.isHovered
 
+  buildPlanClasses: (plan, publishStatus, isPublishing, isActive) ->
     planClasses = [
       'plan'
+      'plan-label-long'
       "#{plan.type}"
       "course-plan-#{plan.id}"
     ]
 
     planClasses.push('is-published') if plan.isPublished or (publishStatus is 'completed')
-    planClasses.push('is-failed') if plan.isFailed
-    planClasses.push('is-killed') if plan.isKilled
+    planClasses.push('is-failed') if publishStatus is 'failed'
+    planClasses.push('is-killed') if publishStatus is 'killed'
     planClasses.push('is-publishing') if isPublishing
     planClasses.push('is-open') if plan.isOpen
     planClasses.push('is-trouble') if plan.isTrouble
+    planClasses.push('active') if isActive
+
+    planClasses.join(' ')
 
-    planClasses = planClasses.join(' ')
+  render: ->
+    {item, courseId} = @props
+    {publishStatus, isPublishing, isHovered, isViewingStats} = @state
+    {plan, displays} = item
+    {durationLength} = plan
 
-    label = @renderLabel(rangeDuration, durationLength, plan, index, offset)
+    planClasses = @buildPlanClasses(plan, publishStatus, isPublishing, isHovered or isViewingStats)
 
-    renderFn = 'renderEditPlan'
-    renderFn = 'renderOpenPlan' if plan.isOpen and plan.isPublished or (isPublishing) or (publishStatus is 'completed')
+    if plan.isPublished or (publishStatus is 'completed')
+      planModal = <CoursePlanDetails
+        plan={plan}
+        courseId={courseId}
+        className={planClasses}
+        ref='details'/>
+    else if isPublishing
+      planModal = <CoursePlanPublishingDetails
+        plan={plan}
+        courseId={courseId}
+        className={planClasses}
+        ref='details'/>
+
+    planDisplays = _.map(displays, (display) =>
+      {rangeDuration, offset, offsetFromPlanStart, index} = display
+
+      labelProps = {rangeDuration, plan, index, offset, offsetFromPlanStart}
+      label = <CoursePlanLabel {...labelProps} ref="label#{index}"/>
+
+      displayComponent = CoursePlanDisplayEdit
+      displayComponent = CoursePlanDisplayQuickLook if planModal?
+
+      displayComponentProps = {
+        plan,
+        display,
+        label,
+        courseId,
+        planModal,
+        planClasses,
+        isFirst: (index is 0),
+        isLast: (index is displays.length - 1),
+        syncHover: @syncHover,
+        removeHover: @removeHover,
+        syncOpenPlan: @syncOpenPlan
+        syncClosePlan: @syncClosePlan
+      }
+
+      <displayComponent
+        {...displayComponentProps}
+        ref="display#{index}"
+        key="display#{index}"/>
+    )
+
+    <div>
+      {planDisplays}
+    </div>
```

Comment:
`render` is getting right at the edge of too long for me. I'd consider splitting it up.  A simple way to do so would be to make a `renderDisplay` method and call change line 174 to be `_.map(displays, => @renderDisplay(...))`.